src/Symfony/Component/Translation/Extractor/PhpAstExtractor.php

```diff
@@ -39,7 +39,11 @@ public function __construct(
             throw new \LogicException(sprintf('You cannot use "%s" as the "nikic/php-parser" package is not installed. Try running "composer require nikic/php-parser".', static::class));
         }
 
-        $this->parser = (new ParserFactory())->create(ParserFactory::PREFER_PHP7);
+        if (method_exists(ParserFactory::class, 'createForVersion')) {
```

Comment:
I suggest checking for the `createForHostVersion` method as that's the one being used.